subsys/mpsl/pm/mpsl_pm_utils.c

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/kernel.h>
+#include <mpsl_pm.h>
+#include <mpsl_pm_config.h>
+#include <zephyr/pm/policy.h>
+#include <zephyr/logging/log.h>
+
+#include <mpsl/mpsl_work.h>
+#include <mpsl/mpsl_pm_utils.h>
+
+LOG_MODULE_REGISTER(mpsl_pm_utils, CONFIG_MPSL_LOG_LEVEL);
+
+/* These values are still arbritrary*/
+#define MAX_DELAY_SINCE_READING_PARAMS 50
+#define TIME_TO_REGISTER_EVENT_IN_ZEPHYR 1000
```

Comment:
```suggestion
/* These constants must be updated once the Zephyr PM Policy API is updated
 * to handle low latency events. Ideally, the Policy API should be changed to use
 * absolute time instead of relative time. This would remove the need for safety
 * margins and allow optimal power savings. */
#define MAX_DELAY_SINCE_READING_PARAMS_US 50
#define TIME_TO_REGISTER_EVENT_IN_ZEPHYR_US 1000
```